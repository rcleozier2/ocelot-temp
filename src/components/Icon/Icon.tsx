import React from "react";

const Icon = (props: any) => {
  return <i className={props.icon} />;
};

export default Icon;
