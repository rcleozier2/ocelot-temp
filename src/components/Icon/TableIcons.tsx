import React from "react";

const Complete = (props: any) => {
  return (
    <>
      <p>
        {props.data.completedOntime} - {props.data.completedLate}
      </p>
    </>
  );
};

const Failed = (props: any) => {
  return (
    <>
      <p>
        {props.data.failedOntime} - {props.data.failedLate}
      </p>
    </>
  );
};

const JustSend = (props: any) => {
  return (
    <>
      <p>{props.data.justSends}</p>
    </>
  );
};

export { Complete, Failed, JustSend };
