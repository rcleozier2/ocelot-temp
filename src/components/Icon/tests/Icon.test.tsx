import React from "react";
import Icon from "../Icon";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const props = {
      icon: "pi pi-thumbs-up"
  }
  const component = renderer
    .create(<Icon  {...props} />)
    .toJSON();
  expect(component).toMatchSnapshot();
});

it("renders correctly with different icon", () => {
    const props = {
        icon: "pi pi-thumbs-down"
    }
    const component = renderer
      .create(<Icon  {...props} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
  