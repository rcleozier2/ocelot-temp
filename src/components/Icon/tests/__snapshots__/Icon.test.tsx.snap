// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<i
  className="pi pi-thumbs-up"
/>
`;

exports[`renders correctly with different icon 1`] = `
<i
  className="pi pi-thumbs-down"
/>
`;
